### ISSUE TITLE
tf.math.tanh produces values strictly smaller than -1

### ISSUE THREAD

**Issue by axch** [tensorflow/tensorflow#34773]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Colab default environment
- TensorFlow installed from (source or binary):
- TensorFlow version (use command below): 1.15.0
- Python version: 2.4.6

**Describe the current behavior**

```
import tensorflow as tf
import numpy as np

tf.enable_v2_behavior()

x = np.float(-8.51089)
tf.math.tanh(x)
=> <tf.Tensor: shape=(), dtype=float32, numpy=-1.0000001>
```
Note the result is less than -1.0.

**Describe the expected behavior**

Expected tanh to stay within the interval [-1.0, 1.0].

**Comment by ravikyram** [comment]
@axch 

I tried in colab with TF 1.15 and i am seeing output (`<tf.Tensor: id=1, shape=(), dtype=float32, numpy=-1.0>`). I tried with TF 2.0, 2.1.0-dev20191202 by commenting (`tf.enable_v2_behavior()`)
i am seeing output(<`tf.Tensor: shape=(), dtype=float32, numpy=-1.0000001>`)Please, find the gist [here.](https://colab.sandbox.google.com/gist/ravikyram/76dbd5a25368435a7beed6ac76152b04/untitled437.ipynb) Is this the expected behavior?. Thanks!

**Comment by axch** [comment]
Interesting.  I had tried it in the internal colab and got -1.0000001 in TF 1.15, but in any case, getting an answer strictly less than -1.0 isn't great in any TF version.

**Comment by ymodak** [comment]
@axch  I see different results than yours. TF 1.15 gives me ```-1.0``` [GitHub gist](https://colab.sandbox.google.com/gist/ymodak/3b1c3949119b9c7bc06d4a0fce8d370b/untitled16.ipynb)
TF 1.15 with GPU ```<tf.Tensor: id=1, shape=(), dtype=float32, numpy=-0.99999994>```

And TF 2.0 outputs ```-1.0```as well [GitHub gist](https://colab.sandbox.google.com/gist/ymodak/e03d2ed7b2bcbd7bdefd2b524b7b29b1/untitled18.ipynb)
TF 2.0 with GPU ```<tf.Tensor: id=1, shape=(), dtype=float32, numpy=-0.99999994>```

**Comment by jburnim** [comment]
I am able to reproduce the issue with TensorFlow 2.1.0-dev20191203 at https://colab.research.google.com/drive/1sGTDyv5K6fNEmX_-j9L89Cb-kUeUi-3F?usp=sharing :
```
!pip install tf-nightly

import numpy as np
import tensorflow as tf

print(tf.__version__)
# ==> 2.1.0-dev20191203

x = np.float32(-8.51089)
tf.math.tanh(x)
# ==> <tf.Tensor: shape=(), dtype=float32, numpy=-1.0000001>
```

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by ymodak** [comment]
This is fixed in tf-nightly version tested with '2.1.0-dev20191226'
